pilot/pkg/networking/core/v1alpha3/cluster_builder.go

```diff
@@ -595,6 +596,7 @@ func (cb *ClusterBuilder) buildLocalityLbEndpoints(proxyNetworkView map[network.
 			LoadBalancingWeight: &wrappers.UInt32Value{
 				Value: instance.Endpoint.GetLoadBalancingWeight(),
 			},
+			HealthStatus: getEndpointHealthStatus(instance),
```

Comment:
I would prefer we never set this field. We can filter out the endpoints instead.

Reply:
Perhaps we can set it based on the WLE `adminState` so we can on purpose disable the traffic to that endpoint.